File: OpenCredentialPublisher.Data/ViewModels/nG/ClrProfileVM.cs
using OpenCredentialPublisher.ClrLibrary.Models;
using OpenCredentialPublisher.Data.Models.ClrEntities;
using System.Collections.Generic;

namespace OpenCredentialPublisher.Data.ViewModels.nG
{
    public class ClrProfileVM
    {
        public string Id { get; set; }
        public string Name { get; set; }
        public string Email { get; set; }
        public string Telephone { get; set; }
        public string StudentId { get; set; }
        public string SourcedId { get; set; }
        public string Url { get; set; }
        public AddressDType Address { get; set; }
        public Dictionary<string, object> AdditionalProperties { get; set; }

        public static ClrProfileVM FromClrProfile(ProfileModel profile)
        {
            if (profile == null)
            {
                return new ClrProfileVM
                {
                    Address = new AddressDType(),
                    AdditionalProperties = new Dictionary<string, object>()
                };
            }
            else
            {
                return new ClrProfileVM
                {
                    Address = profile.Address,
                    AdditionalProperties = profile.AdditionalProperties,
                    Email = profile.Email,
                    Telephone = profile.Telephone,
                    Id = profile.Id,
                    Name = profile.Name,
                    SourcedId = profile.SourcedId,
                    StudentId = profile.StudentId,
                    Url = profile.Url
                };
            }
        }

    }
}

File: OpenCredentialPublisher.Wallet/src/ClientApp/src/app/shared/models/clrProfileVM.ts
import { AddressDType } from "./clrLibrary/addressDType";
import { ProfileDType } from "./clrLibrary/profileDType";

export class ClrProfileVM {
  id: string;
  name: string;
  email: string;
  telephone: string;
  studentId: string;
  sourcedId: string;
  url: string;
  address: AddressDType;
  additionalProperties: { [index: string]: any };
  constructor(profile: ProfileDType){
    if (profile == null)
    {
      this.address = new AddressDType();
      this.additionalProperties = {}
      this.email = '';
      this.telephone = '';
      this.id = '';
      this.name = '';
      this.sourcedId = '';
      this.studentId = '';
      this.url = '';
    }
    else
    {
      this.address = profile.address;
      this.additionalProperties = profile.additionalProperties;
      this.email = profile.email;
      this.telephone = profile.telephone;
      this.id = profile.id;
      this.name = profile.name;
      this.sourcedId = profile.sourcedId;
      this.studentId = profile.studentId;
      this.url = profile.url;
    }
  }
}
